tests/update.rs

```diff
@@ -17,83 +17,110 @@ fn mock_substrate_stdlib() -> Vec<u8> {
 }
 
+fn verify_module_error_with_msg(
+    res: DispatchResultWithPostInfo,
+    e_msg: &str,
+) -> Result<bool, String> {
+    let err = if let Err(err) = res {
+        err
+    } else {
+        return Err(format!("{res:?} is not an error"));
+    };
+    let err = if let DispatchError::Module(mod_err) = err.error {
+        mod_err
+    } else {
+        return Err(format!("{err:?} is not a ModuleError"));
+    };
+    if let Some(msg) = err.message {
+        Ok(msg == e_msg)
+    } else {
+        Err(format!("{err:?} contains no message"))
+    }
+}
+
+macro_rules! move_script_transaction {
+    ($bytecode:expr, $type_args:expr, $($args:expr),*) => {
+        {
+            let transaction = ScriptTransaction {
+                bytecode: $bytecode,
+                type_args: $type_args,
+                args: vec![$(bcs::to_bytes($args).unwrap()),*],
+            };
+            bcs::to_bytes(&transaction).unwrap()
+        }
+    }
+}
+
 #[test]
-fn regular_user_fail() {
-    let move_stdlib = move_stdlib_bundle().to_vec();
-    let sub_stdlib = substrate_stdlib_bundle().to_vec();
+fn regular_user_update_fail() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_err!(
+            MoveModule::update_stdlib_bundle(
+                RuntimeOrigin::signed(CAFE_ADDR_NATIVE.clone()),
+                mock_move_stdlib(),
+            ),
+            DispatchError::BadOrigin,
+        );
+        assert_err!(
+            MoveModule::update_stdlib_bundle(
+                RuntimeOrigin::signed(CAFE_ADDR_NATIVE.clone()),
+                mock_substrate_stdlib(),
+            ),
+            DispatchError::BadOrigin,
+        );
+    });
+}
+
+#[test]
+fn change_interface_add_param_fail() {
+    ExtBuilder::default().build().execute_with(|| {
+        let res = MoveModule::update_stdlib_bundle(RuntimeOrigin::root(), mock_move_stdlib());
+        assert!(verify_module_error_with_msg(res, "BackwardIncompatibleModuleUpdate").unwrap());
+    });
+}
 
+#[test]
+fn change_interface_remove_param_fail() {
     ExtBuilder::default()
         .with_move_stdlib(Some(mock_move_stdlib()))
         .with_substrate_stdlib(Some(mock_substrate_stdlib()))
         .build()
         .execute_with(|| {
-            assert_err!(
-                MoveModule::update_stdlib(
-                    RuntimeOrigin::signed(CAFE_ADDR_NATIVE.clone()),
-                    move_stdlib,
-                ),
-                DispatchError::BadOrigin,
-            );
-            assert_err!(
-                MoveModule::update_stdlib(
-                    RuntimeOrigin::signed(CAFE_ADDR_NATIVE.clone()),
-                    sub_stdlib,
-                ),
-                DispatchError::BadOrigin,
+            let res = MoveModule::update_stdlib_bundle(
+                RuntimeOrigin::root(),
+                move_stdlib_bundle().to_vec(),
             );
+            assert!(verify_module_error_with_msg(res, "BackwardIncompatibleModuleUpdate").unwrap());
```

Comment:
General comment - The file should be named `stdlib-update.rs`

Reply:
Done